### PR TITLE
Use rank to order Propositions and expose priority property

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Description

- MOB-22213 - Responses from IDS are now sorted by their `rank` values in the Proposition item
- MOB-22248 - `scopeDetails.activity.priority` field now exposed in `Proposition` and `ContentCardUI` classes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPMessaging.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2773,7 +2785,7 @@
 					"@executable_path/Frameworks",
 				);
 				MARKETING_VERSION = 1.0;
-				PRODUCT_BUNDLE_IDENTIFIER = com.adobe.MessagingDemoAppSwiftUI;
+				PRODUCT_BUNDLE_IDENTIFIER = com.steveb.priorityTester;
```
Comment (PravinPK) [inline AEPMessaging.xcodeproj/project.pbxproj:2788]: haha..I guess we both will be switching this back and forth

File: AEPMessaging/Sources/Messaging+State.swift [hunk 2]
```diff
@@ -55,10 +55,7 @@ extension Messaging {
     private func hydratePropositionsRulesEngine() {
         let parsedPropositions = ParsedPropositions(with: inMemoryPropositions, requestedSurfaces: inMemoryPropositions.map { $0.key }, runtime: runtime)
         if let inAppRules = parsedPropositions.surfaceRulesBySchemaType[.inapp] {
-            /// MOB-21852 - iam items are returned in reverse priority order, so we need to flip the
-            /// order of the array prior to saving them and hydrating the rules engine.  this allows the
-            /// highest priority item to be shown first when rules engine evaluates top-down
-            rulesEngine.launchRulesEngine.replaceRules(with: inAppRules.flatMap { $0.value.reversed() })
+            rulesEngine.launchRulesEngine.replaceRules(with: inAppRules.flatMap { $0.value })
```
Comment (PravinPK) [inline AEPMessaging/Sources/Messaging+State.swift:58]: so much relieved to see this getting flipped

File: TestApps/MessagingDemoAppSwiftUI/AppPages/CardsView.swift [hunk 3]
```diff
@@ -71,7 +71,8 @@ struct CardsView: View, ContentCardUIEventListening {
             showLoadingIndicator = false
             switch result {
             case .success(let cards):
-                savedCards = cards
+                // sort the cards by priority order and save them to our state property
+                savedCards = cards.sorted { $0.priority > $1.priority }
```
[the remark this answers is not in the document]
Comment (PravinPK) [reply TestApps/MessagingDemoAppSwiftUI/AppPages/CardsView.swift:75]: We can do them right here : https://github.com/adobe/aepsdk-messaging-ios/blob/dev-v5.5.0/AEPMessaging/Sources/UI/Messaging%2BUIPublicAPI.swift#L58

**Review by PravinPK — APPROVED** [review]
Everything looks good except for one small request

File: TestApps/MessagingDemoAppSwiftUI/AppPages/CardsView.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply TestApps/MessagingDemoAppSwiftUI/AppPages/CardsView.swift:75]: we chatted offline and decided to keep the source in its relative "qualification" order for this session.

in the future, we'll likely need to have other options for sorting - until then we'll just leave it as it is

**Pull request merged**